src/components/Asset/AssetActions/Trade/FormTrade.tsx

```diff
@@ -64,32 +73,70 @@ export default function FormTrade({
     .defined()
 
   async function handleTrade(values: FormTradeData) {
+    if (!web3 || !asset || !poolInfo || !values) return
+
     try {
+      const poolInstance = new Pool(web3)
+
+      const swapMarketFeeIn = await poolInstance.getMarketFees(
+        asset.accessDetails.addressOrId,
+        poolInfo.baseTokenAddress
+      )
+
+      const swapMarketFeeOut = await poolInstance.getMarketFees(
+        asset.accessDetails.addressOrId,
+        poolInfo.datatokenAddress
+      )
+
+      const tokenInOutMarket: TokenInOutMarket = {
+        tokenIn: poolInfo.baseTokenAddress,
+        tokenOut: poolInfo.datatokenAddress,
+        marketFeeAddress: appConfig.consumeMarketPoolSwapFee
+      }
+      const tokenOutMarket: TokenInOutMarket = {
+        tokenIn: poolInfo.datatokenAddress,
+        tokenOut: poolInfo.baseTokenAddress,
+        marketFeeAddress: appConfig.consumeMarketPoolSwapFee
+      }
+
       const impact = new Decimal(
         new Decimal(100).sub(new Decimal(values.slippage))
       ).div(100)
       const precision = 15
```

Comment:
oh boy, great leftover. Anybody knows a reason why we specifically were using `15` here? Cause in all other files we do something like `Decimal.set({ precision: 18 })` and not redefine that within the file

Reply:
i'm not sure if we can set this globally, that would be great. Needs some testing, i'll create a new issue